### ISSUE TITLE
Is SSL Pinning possible with Kotlin Multi Platform

### ISSUE THREAD

**Issue by boldijar** [apollographql/apollo-kotlin#5736]
### Question

I need to implement SSL pinning especially in iOS, on android is pretty simple since okhttp can do all this, but ios is using somehow NSUrlSession and I’m not sure if is possible to implement SSL Pinning.
https://slack-chats.kotlinlang.org/t/541377/is-there-a-way-to-do-certificate-pinning-on-the-apollo-kotli
I’ve tried to read this thread from 2 years ago but it was no luck yet.

I’m thinking that the certificate should be somewhere in Xcode project, and somehow I can build the SSL Pinning from the DefaultHttpEngine in ios library project? 
If anybody tried something like this please let me know.

**Comment by martinbonnin** [comment]
Since that KotlinLang thread, Apollo now ships with `apollo-engine-ktor` so if you can do KMP certificate pinning with ktor, you can probably do so with Apollo.

**Comment by boldijar** [comment]
> Since that KotlinLang thread, Apollo now ships with `apollo-engine-ktor` so if you can do KMP certificate pinning with ktor, you can probably do so with Apollo.

This is only in v4.0.0-alpha.3 right?
And I assume at some point it will become a stable 4.0.0, right?

**Comment by martinbonnin** [comment]
> This is only in v4.0.0-alpha.3 right?

Yes it is. Although you can copy/paste the implementation in 3.x if needed. The `HttpEngine` interface is relatively small.

> And I assume at some point it will become a stable 4.0.0, right?

Yup, current goal is around KotlinConf but no promises. Also while the API is not 100% stable, the betas are pretty stable in terms of functionality so unless you're shiping a lib, they're pretty safe to use.
